migration/iD12/oracle/202406181038_IDEMPIERE-6177.sql

```diff
@@ -0,0 +1,81 @@
+;
+
+-- Jun 18, 2024, 10:43:12 AM CEST
+ALTER TABLE C_DocType ADD IsAutoGenerateInout CHAR(1) DEFAULT 'N' CHECK (IsAutoGenerateInout IN ('Y','N')) NOT NULL
+;
+
+-- Jun 18, 2024, 10:43:27 AM CEST
+ALTER TABLE C_DocType ADD IsAutoGenerateInvoice CHAR(1) DEFAULT 'N' CHECK (IsAutoGenerateInvoice IN ('Y','N')) NOT NULL
+;
+
+-- Jun 18, 2024, 10:44:03 AM CEST
+INSERT INTO AD_Field (AD_Field_ID,Name,Description,Help,AD_Tab_ID,AD_Column_ID,IsDisplayed,DisplayLength,SeqNo,IsSameLine,IsHeading,IsFieldOnly,IsEncrypted,AD_Client_ID,AD_Org_ID,IsActive,Created,CreatedBy,Updated,UpdatedBy,IsReadOnly,IsCentrallyMaintained,EntityType,AD_Field_UU,IsDisplayedGrid,SeqNoGrid,XPosition,ColumnSpan) VALUES (208492,'Generate Invoice','Automatically Generate Invoice after (Sales/Purchase) Order completed','When selected the Order will generate Invoice automatically. Note that only Orders with document subtype Standard and Prepay Order can be applied!',167,216612,'Y',1,370,'N','N','N','N',0,0,'Y',TO_TIMESTAMP('2024-06-18 10:44:02','YYYY-MM-DD HH24:MI:SS'),100,TO_TIMESTAMP('2024-06-18 10:44:02','YYYY-MM-DD HH24:MI:SS'),100,'N','Y','D','78a968cd-135c-42a3-9da7-a75057cc703d','Y',330,2,2)
+;
+
+-- Jun 18, 2024, 10:44:03 AM CEST
+INSERT INTO AD_Field (AD_Field_ID,Name,Description,Help,AD_Tab_ID,AD_Column_ID,IsDisplayed,DisplayLength,SeqNo,IsSameLine,IsHeading,IsFieldOnly,IsEncrypted,AD_Client_ID,AD_Org_ID,IsActive,Created,CreatedBy,Updated,UpdatedBy,IsReadOnly,IsCentrallyMaintained,EntityType,AD_Field_UU,IsDisplayedGrid,SeqNoGrid,XPosition,ColumnSpan) VALUES (208493,'Generate Shipment','Automatically Generate Shipment after (Sales/Purchase) Order completed','When selected the Order will generate Shipment automatically. Note that only Orders with document subtype Standard and Prepay Order can be applied!',167,216613,'Y',1,380,'N','N','N','N',0,0,'Y',TO_TIMESTAMP('2024-06-18 10:44:03','YYYY-MM-DD HH24:MI:SS'),100,TO_TIMESTAMP('2024-06-18 10:44:03','YYYY-MM-DD HH24:MI:SS'),100,'N','Y','D','fbb23049-e37b-40ad-b414-93b789dff9ed','Y',340,2,2)
+;
+
+-- Jun 18, 2024, 10:46:25 AM CEST
+UPDATE AD_Field SET SeqNo=330,Updated=TO_TIMESTAMP('2024-06-18 10:46:25','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Field_ID=6567
+;
+
+-- Jun 18, 2024, 10:46:25 AM CEST
+UPDATE AD_Field SET SeqNo=340,Updated=TO_TIMESTAMP('2024-06-18 10:46:25','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Field_ID=3125
+;
+
+-- Jun 18, 2024, 10:46:25 AM CEST
+UPDATE AD_Field SET DisplayLogic='@DocBaseType@=''SOO'' | @DocBaseType@=''POO''', SeqNo=350,Updated=TO_TIMESTAMP('2024-06-18 10:46:25','YYYY-MM-DD HH24:MI:SS'),UpdatedBy=100 WHERE AD_Field_ID=208492
```

Comment:
from what I see here, the two new flags are visible for Sales and Purchase Orders, independent from the SubType, but the code just works for:

invoice -> POS, OnCredit, Prepay
shipment -> POS, OnCredit, Prepay, Warehouse

I think we can have two approaches here ( would like also opinion from @hengsin ):

1)  keep the flags just for  those 3-4 subtypes, for that we need:
1a) MDocType.beforeSave must take care of setting the flags to N on the other types
1b) DisplayLogic must show those flags just for the intended subtypes

or

2) we can allow the flags for any type of sales order (including Standard)
2a) in such case the validations on completeIt about DocSubTypeSO would be unnecessary and we need to check JUST the flag
2b) we need a migration script that sets the flags for backward compatibility:
AutoGenInv=Y for POS, OnCredit, Prepay
AutoGenShip=Y for POS, OnCredit, Prepay, Warehouse

this second case will make all the DocSubType documentation basically obsolete and indeed the concept of DocSubType would be obsolete, as we can model all of that just with Standard Orders and the flags.

WDYT?  Which approach would be better?

Reply:
2 is very disruptive, probably safer to go with 1 instead.

Reply:
OK, so @PeterTakacs300 - we need the following changes here:

1a) MDocType.beforeSave must take care of setting the flags to N on the other types
1b) DisplayLogic must show those flags just for the intended subtypes
invoice -> POS, OnCredit, Prepay
shipment -> POS, OnCredit, Prepay, Warehouse

Reply:
Done - but need to check if there is not a collision now with the code, if this feature is intended to work with all the listed subtypes, or just with Prepay Order

Reply:
Consulted with Igor, this feature was intended only for Prepay Order for now, could be problematic for the other subtypes. In the latest commit I've modified the beforeSave and display logic accordingly.